TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,14 +145,127 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
 
         public async Task<CommandResult> Pokedex(CommandContext context)
         {
-            User user = (await context.ParseArgs<Optional<User>>()).OrElse(context.Message.User);
+            (Optional<User> optionalUser, Optional<string> optionalMode, Optional<User> optionalCompareUser) =
+                await context.ParseArgs<Optional<User>, Optional<string>, Optional<User>>();
+            User user = optionalUser.IsPresent ? optionalUser.Value : context.Message.User;
             bool isSelf = user == context.Message.User;
-            int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
+
+            if (!optionalMode.IsPresent)
+            {
+                int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
+                return new CommandResult
+                {
+                    Response = isSelf
+                        ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                        : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                };
+            }
+
+            string mode = optionalMode.Value;
+            ImmutableSortedDictionary<PkmnSpecies, int> numBadgesPerSpecies =
+                await _badgeRepo.CountByUserPerSpecies(user.Id);
+
+            if (mode.Equals(PokedexModeMissing))
+            {
+                IEnumerable<PkmnSpecies> missingList = _KnownSpecies.Except(numBadgesPerSpecies.Keys);
+                // Seems like PokedexData is not sorted. So we have to sort the list.
```

Comment:
So you want to change this part? `
        private PokedexData()
        {
            KnownSpecies = LoadSpeciesNames().ToImmutableHashSet();
        }`

Unfortunately I'm unsure about which Collection types should be used in which situations. I just wrote the comment there, but I guess it makes sense that a HashSet is not sorted in any way

Reply:
can I just paste patches as comments and github can apply them...?
```patch
Index: TPP.Core/PokedexData.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/TPP.Core/PokedexData.cs b/TPP.Core/PokedexData.cs
--- a/TPP.Core/PokedexData.cs	(revision 24f7494677330df2e1e20dddf920388786848582)
+++ b/TPP.Core/PokedexData.cs	(date 1615568330788)
@@ -11,7 +11,7 @@
 {
     public class PokedexData
     {
-        public IImmutableSet<PkmnSpecies> KnownSpecies { get; }
+        public ImmutableSortedSet<PkmnSpecies> KnownSpecies { get; }
 
         private static readonly (string, string)[] NameNormalizations =
         {
@@ -58,7 +58,7 @@
 
         private PokedexData()
         {
-            KnownSpecies = LoadSpeciesNames().ToImmutableHashSet();
+            KnownSpecies = LoadSpeciesNames().ToImmutableSortedSet();
         }
 
         /// Construct an instance with all the pokedex data loaded,
```

Reply:
aw only syntax highlighting

Reply:
I have checked all usages/occurances of PokedexData and PokedexData.KnownSpecies (most of them were in testing modules tbf) 
Did not find anything that would speak against changing the datatype. Also I removed the "OrderBy" at "!dex missing" because it actually caused Fakemons to get printed all over the place